### PR TITLE
Fix: model id and model dir check order

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
Fix the condition check order, we should always check if `model_id` is None first as `model_dir` can be always specified.
Default `--model_dir` to `/mnt/models`, so user does not need to add this additional argument on inference service yaml. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Special notes for your reviewer**:

1. This depends on https://github.com/kserve/kserve/pull/3679 which should be merged first

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [X] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]
/lgtm
/hold

File: python/huggingfaceserver/huggingfaceserver/__main__.py [hunk 1]
```diff
@@ -52,10 +52,12 @@ def list_of_strings(arg):
 parser.add_argument(
     "--model_dir",
     required=False,
-    default=None,
+    default="/mnt/models",
```
Comment (ghost) [inline python/huggingfaceserver/huggingfaceserver/__main__.py:55]: Wouldn't this be problematic as [L136](https://github.com/kserve/kserve/pull/3680/files#diff-2ed97434a3b43aac01ea6d9f2364d6c176a3716fcff260115226a61260d460f9L133)/[L156](https://github.com/kserve/kserve/blob/c660972ea153a197a483efb23b5c5e036eb4504c/python/huggingfaceserver/huggingfaceserver/__main__.py#L156) would always set `args.model` to `/mnt/models`?
Should the line be then updated to `args.model = args.model_id or args.model_dir`

**Pull request merged**